vue2-vuetify/src/controls/StringMaskControlRenderer.vue

```diff
@@ -0,0 +1,292 @@
+    oldVnode: VNode
+  ) => {
+    if (this.delegate?.update) {
+      this.delegate?.update(el, binding, vnode, oldVnode);
+    }
+  };
+
+  componentUpdated = (
+    el: HTMLElement,
+    binding: DirectiveBinding,
+    vnode: VNode,
+    oldVnode: VNode
+  ) => {
+    if (this.delegate?.componentUpdated) {
+      this.delegate?.componentUpdated(el, binding, vnode, oldVnode);
+    }
+  };
+
+  unbind = (
+    el: HTMLElement,
+    binding: DirectiveBinding,
+    vnode: VNode,
+    oldVnode: VNode
+  ) => {
+    if (this.delegate?.unbind) {
+      this.delegate?.unbind(el, binding, vnode, oldVnode);
+    }
+    this.delegate = undefined;
+  };
+}
+
+const controlRenderer = defineComponent({
+  name: 'string-mask-control-renderer',
+  components: {
+    ControlWrapper,
+    VHover,
+    VTextField,
+  },
+  directives: {
+    DisabledIconFocus,
+    Mask: new DelegateDirective(),
+  },
+  props: {
+    ...rendererProps<ControlElement>(),
+  },
+  setup(props: RendererProps<ControlElement>) {
+    const adaptValue = (value: any) => value || undefined;
+    const control = useVuetifyControl(useJsonFormsControl(props), adaptValue);
+
+    return { ...control, adaptValue };
+  },
+  methods: {
+    maskedValue(value: string | undefined): string | undefined {
+      if (this.mask && !this.returnMaskedValue) {
+        return this.maskFilter(value, this.mask);
+      }
+
+      return value;
+    },
+    unmaskedValue(value: string | undefined): string | undefined {
+      if (this.mask && !this.returnMaskedValue && value) {
+        value = value
+          .split('')
+          .map((char, index) => {
+            if (this.mask.length > index) {
+              const replacer = this.maskReplacers[this.mask[index]];
+              // #, A, N, X are default unless the replacer is null (override)
+              return (['#', 'A', 'N', 'X'].includes(this.mask[index]) &&
+                replacer === undefined) ||
+                replacer
+                ? char
+                : '';
+            }
+            return char;
+          })
+          .join('');
+      }
+      return value;
+    },
+  },
+  computed: {
+    maskModel: {
+      get(): string | undefined {
+        let value = this.control.data;
+        if (this.mask && !this.returnMaskedValue && value) {
+          value = this.maskedValue(value);
+        }
+        return value;
+      },
+      set(val: string | undefined): void {
+        let value = val;
+
+        if (this.mask) {
+          if (this.mask && !this.returnMaskedValue) {
+            value = this.unmaskedValue(value);
+          }
+          if (this.adaptValue(value) !== this.control.data) {
+            // only invoke onChange when values are different since v-mask is also listening on input which lead to loop
+
+            this.onChange(value);
+          }
+        } else {
+          this.onChange(value);
+        }
+      },
+    },
+    mask(): string {
+      return typeof this.appliedOptions.mask === 'string'
```

Comment:
sure thing - the if checks were there because initially the code were in the String renderer but I decided to not complicate that one and have a separate one